### PR TITLE
Fix/dl stop syncing

### PR DESCRIPTION
**Pull request by whilei**
frankensteined from ethereum/go-ethereum#14460

### CODE REVIEW

File: eth/downloader/peer.go [hunk 1]
```diff
@@ -386,9 +379,8 @@ func (ps *peerSet) Register(p *peer) error {
 
 	// Register the new peer with some meaningful defaults
 	ps.lock.Lock()
-	defer ps.lock.Unlock()
```
Comment (tzdybal) [inline eth/downloader/peer.go]: What's the rationale behind this change? Code before change is pretty equivalent?

Comment (whilei) [reply eth/downloader/peer.go]: IDK - came with the merge. Hadn't looked closely. I prefer the `defer` though ;)
